llvm/lib/CodeGen/ShrinkWrap.cpp

```diff
@@ -957,6 +957,15 @@ bool ShrinkWrap::runOnMachineFunction(MachineFunction &MF) {
   if (!ArePointsInteresting())
     return Changed;
 
+  const TargetLowering *TLI = MF.getSubtarget().getTargetLowering();
+  while (TLI->hasInlineStackProbe(MF) &&
```

Comment:
`TLI->hasInlineStackProbe(MF)` is a loop invariant, no need to be in the condition.